Src/WitsmlExplorer.Frontend/components/Modals/AnalyzeGapModal.tsx

```diff
@@ -0,0 +1,114 @@
+  const [gapSizeShowWarning, setGapSizeShowWarning] = useState<boolean>(false);
+
+  const onSubmit = async () => {
+    setIsLoading(true);
+    const logReference: LogObject = props.logObject;
+    const analyzeGapsJob = {
+      logReference: logReference,
+      mnemonics: props.mnemonics,
+      gapSize: gapSize,
+      timeGapSize: convertToMilliseconds(timeGapSize)
+    };
+    const jobId = await JobService.orderJob(JobType.AnalyzeGaps, analyzeGapsJob);
+    const reportModalProps = { jobId };
+    setIsLoading(false);
+    dispatchOperation({ type: OperationType.HideModal });
+    dispatchOperation({ type: OperationType.DisplayModal, payload: <ReportModal {...reportModalProps} /> });
+  };
+
+  const handleGapSizeChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.value) {
+      const value = Number(event.target.value);
+      setGapSize(value);
+      setGapSizeIsValid(value > 0);
+      setGapSizeShowWarning(value >= 0 && value < 1);
+    }
+  };
+
+  const handleTimeGapSizeChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setGapSizeIsValid(new RegExp(timePattern).test(event.target.value));
+    if (event.target.value) {
+      setTimeGapSize(event.target.value);
+      const millisecondsTime = convertToMilliseconds(event.target.value);
+      setGapSizeIsValid(millisecondsTime > 0);
+      setGapSizeShowWarning(millisecondsTime > 0 && millisecondsTime < 60000);
+    }
+  };
+
+  const convertToMilliseconds = (time: string): number => {
+    const [hours, minutes, seconds] = time.split(":").map(Number);
+    return (hours * 3600 + minutes * 60 + seconds) * 1000;
+  };
+
+  return (
+    <>
+      {log && (
+        <ModalDialog
+          heading={`Analyze gaps`}
+          content={
+            <>
+              {log.indexType === WITSML_INDEX_TYPE_DATE_TIME && (
+                <TextField
+                  id={"timeGapSize"}
+                  label="gap size"
+                  value={timeGapSize}
+                  onChange={handleTimeGapSizeChanged}
+                  placeholder="hh:mm:ss"
+                  maxLength={8}
+                  helperText={
+                    !gapSizeIsValid ? "Gap size time must be in format hh:mm:ss, where hours: 00..23, minutes: 00..59, seconds: 00..59 and must be greater then 00:00:00" : null
+                  }
+                />
+              )}
+              {log.indexType === WITSML_INDEX_TYPE_MD && (
+                <TextField
+                  id={"gapSize"}
+                  label="gap size"
+                  type="number"
+                  value={gapSize}
+                  onChange={handleGapSizeChanged}
+                  helperText={!gapSizeIsValid ? "Gap size number must be greater than 0" : null}
+                />
+              )}
+              {gapSizeShowWarning && <WarningBar message={"The gap size is too small, gap analysis might take some time."} />}
```

Comment:
Does the gap size actually change the duration of the analysis? It checks the indexes between all data rows, so the gap size would only take the time extra that it takes to add it to the report, right? 

It could matter if you added some "smart" checking that skips rows based on the gap size, and only looks deeper if the difference between the skipped parts is too big, in a "divide and conquer"-style, but that might be a bit overkill for this task.

Reply:
It is exactly as you write, the gap size would only take the time extra that it takes to add it to the report. Some "smart" checking can be improved in the future if needed.